apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1.Tests/StorageClientImplTest.cs

```diff
@@ -69,5 +76,418 @@ public void GetEffectiveEncryptionKey_Invalid(EncryptionKey clientkey, Encryptio
+   NoRetryHelper(service => service.Buckets.TestIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })), client => client.TestBucketIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })));
+
+        [Fact]
+        public void TestBucketIamPolicy_RetryOnce() =>
+RetryOnceHelper(service => service.Buckets.TestIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })), client => client.TestBucketIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })));
+
+        [Fact]
+        public void TestBucketIamPolicy_RetryThenFail() =>
+    RetryThenFailHelper(service => service.Buckets.TestIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })), client => client.TestBucketIamPermissions("bucket", new Repeatable<string>(new[] { "permission" })));
+
+        [Fact]
+        public void LockRetentionPolicy_NoRetry() =>
+   NoRetryHelper(service => service.Buckets.LockRetentionPolicy("bucket", 70), client => client.LockBucketRetentionPolicy("bucket", 70));
+
+        [Fact]
+        public void LockRetentionPolicy_RetryOnce() =>
+   RetryOnceHelper(service => service.Buckets.LockRetentionPolicy("bucket", 70), client => client.LockBucketRetentionPolicy("bucket", 70));
+
+        [Fact]
+        public void LockRetentionPolicy_RetryThenFail() =>
+    RetryThenFailHelper(service => service.Buckets.LockRetentionPolicy("bucket", 70), client => client.LockBucketRetentionPolicy("bucket", 70));
+
+        // IAM test cases END
+
+        // Object test cases BEGIN
+
+        [Fact]
+        public void GetObject_NoRetry() =>
+            NoRetryHelper(service => service.Objects.Get("bucket", "object"), client => client.GetObject("bucket", "object"));
+
+        [Fact]
+        public void GetObject_RetryOnce() =>
+            RetryOnceHelper(service => service.Objects.Get("bucket", "object"), client => client.GetObject("bucket", "object"));
+
+        [Fact]
+        public void GetObject_RetryThenFail() =>
+            RetryThenFailHelper(service => service.Objects.Get("bucket", "object"), client => client.GetObject("bucket", "object"));
+
+
+        /*
+         * /copy object test cases are currently failing
+         * Also not sure where to add  IfGenerationMatch condition to test
+         * Error:
+         Assert.Equal() Failure
+        Expected: https://storage.googleapis.com/storage/v1/b/sourceBucket/o/sourceObject/copyTo/b/destinationBucket/o/destinationObject
+        Actual:   https://storage.googleapis.com/storage/v1/b/sourceBucket/o/sourceObject/rewriteTo/b/destinationBucket/o/destinationObject
+         */
+
+        [Fact]
+        public void CopyObject_NoRetry() =>
+      NoRetryHelper(service => service.Objects.Copy(new Object() { Id = "id", Name = "name" }, "sourceBucket", "sourceObject", "destinationBucket", "destinationObject"), client => client.CopyObject("sourceBucket", "sourceObject", "destinationBucket", "destinationObject"));
```

Comment:
We're calling RewriteObject the API to implement object copying, so this should be `service.Objects.RewriteObject(...)`

Not sure what you mean about "not sure where to add IfGenerationMatch condition to test" - that should be basically in the same way as other cases. (To specify it in the request you'll need `new ObjectsResource.RewriteRequest` instead of `service.Objects.RewriteObject`, so that you can use an object initializer.)